src/github/GithubClient.ts

```diff
@@ -0,0 +1,368 @@
+    core.debug(`Dir: ${JSON.stringify(fs.readdirSync(rootDirectory))}`);
+
+    const info = await suppressOutput(async () =>
+      client.uploadArtifact(name, [file], rootDirectory, {
+        continueOnError: false,
+      })
+    );
+
+    core.debug("uploadArtifact response:");
+    core.debug(JSON.stringify(info));
+  }
+
+  // --------------- COMPLETED WORKFLOW METHODS ------------------
+
+  /**
+   * Lists the workflow run artifacts for a completed workflow
+   * @param runId the workflow run number
+   */
+  async listWorkflowRunArtifacts({
+    runId,
+  }: {
+    runId: number;
+  }): Promise<(Artifact & { id: number })[]> {
+    const response = await this.client.rest.actions.listWorkflowRunArtifacts({
+      ...this.repo,
+      run_id: runId,
+      per_page: 100,
+      page: 1,
+    });
+
+    core.debug(dashWrap("listWorkflowRunArtifacts"));
+    core.debug(JSON.stringify(response));
+
+    if (response.status >= 400) {
+      throw new Error("Unable to retrieve listWorkflowRunArtifacts");
+    }
+
+    return response.data.artifacts;
+  }
+
+  /**
+   * Lists the workflow run artifacts for a completed workflow
+   * @param runId the workflow run number
+   */
+  async findLatestWorkflowRunForBranch({
+    branch,
+  }: {
+    branch: string;
+  }): Promise<WorkflowRun> {
+    const response = await this.client.rest.actions.listWorkflowRunsForRepo({
+      ...this.repo,
+      branch,
+      status: "success",
+      per_page: 100,
+      page: 1,
+    });
+
+    core.debug(dashWrap("findLatestWorkflowRunForBranch"));
+    core.debug(JSON.stringify(response));
+
+    if (response.status >= 400) {
+      throw new Error("Unable to findLatestWorkflowRunForBranch");
+    }
+
+    return response.data.workflow_runs[0];
+  }
+
+  /**
+   * Downloads the artifact and returns a reference to the file
+   * @param artifactId the artifact id to download
+   */
+  async downloadWorkflowRunArtifact({
+    artifactId,
+  }: {
+    artifactId: number;
+  }): Promise<string> {
+    const response = await this.client.rest.actions.downloadArtifact({
+      ...this.repo,
+      artifact_id: artifactId,
+      archive_format: "zip",
+    });
+
+    core.debug("downloadArtifact response:");
+    core.debug(JSON.stringify(response));
+
+    const artifactZip = await cache.downloadTool(response.url);
+
+    core.debug("downloadTool response:");
+    core.debug(JSON.stringify(artifactZip));
+
+    const artifactPath = await cache.extractZip(artifactZip);
+
+    core.debug("extractZip response:");
+    core.debug(JSON.stringify(artifactPath));
+
+    for (const file of fs.readdirSync(artifactPath)) {
+      const filePath = `${artifactPath}/${file}`;
+      if (fs.existsSync(filePath)) {
+        return filePath;
+      }
+    }
+
+    return "";
+  }
+
+  // --------------- RELEASE ASSET METHODS ------------------
+
+  /**
+   * Uploads a release asset
+   * @param release release object
+   * @param fileName name of the asset
```

Comment:
You mentioned on the call renaming fileName to assetName. Is that still a consideration or is fileName good for now?